### ISSUE TITLE
Notate 'introduction' with anacrusis?

### ISSUE THREAD

**Issue by bennorth** [raspberrypilearning/code-a-carol#6]
Great project!  Hope it's OK to make a suggestion:

In the first music notation in [step 4](https://projects.raspberrypi.org/en/projects/code-a-carol/4) ([en/images/introduction.png](https://github.com/raspberrypilearning/code-a-carol/blob/master/en/images/introduction.png)), I think the bar divisions are in the wrong place.  I think the melody should start on beat 4 of a bar (as, e.g., [here](https://www.christmasmusicsongs.com/twelve-days-of-christmas-sheet-music.html)).  I haven't created a PR, sorry, as I don't know what system you've used to create the PNGs.

**Comment by martinohanlon** [comment]
@lawsie can you point me to what software you used to create the notation? e.g. https://github.com/raspberrypilearning/code-a-carol/blob/master/en/images/introduction.png

**Comment by lawsie** [comment]
It was http://flat.io that I used for the notation. 

I'm not sure how much musical knowledge you have @martinohanlon but for info, the key of the music that is linked above is F, whereas in my notation I've done something deliberately which is a bit naughty - I've not put in a key signature (i.e. left as C Major) but put in the B flats where they occur, so that people who don't know how to read music can see them. This means (for you) that any notation in the linked page above will NOT show a B flat where it occurs. The linked music is also different in that (for example) there are no rests between "my true love gave to me" and "a partridge in a pear tree", so if you change to the version above you'll need to add in a rest at the end of the 3rd bar to make it whole, this will change the code, and there will be a slight knock on effect on how the tune generated by the project sounds if all other images/code on other steps are left as they are.

Happy to have some more musical advice though @bennorth as it's been a while since I did anything properly musical.

**Comment by bennorth** [comment]
Thanks for looking at this very non-urgent suggestion!

I had a bit of a look at flat.io; nice.  In my quick play, though, I couldn't get the spacing of the anacrusis to look right.  For an experiment, I created the attached PDF in [MuseScore](https://musescore.org/), which I have a tiny bit of experience with.  (It's a GPL'd piece of software, which fits well with the Raspberry Pi philosophy of being able to learn how things work.)

[Twelve-days-fragments.pdf](https://github.com/raspberrypilearning/code-a-carol/files/2760499/Twelve-days-fragments.pdf)

I made it just have the treble clef, as there was nothing in the bass clef in any of the snippets.  The change back to 4/4 on p.3 (the 'Four, three, two' snippet) looks odd and there are only three beats in it, but that bar leads straight into 'and a partridge etc.' so I _think_ the time-signature change should be there.  The 'Item' snippet (p.4) looks a bit stretched; I can look into how to scrunch it up a bit.

If you think this looks OK, I can create some PNGs from it.  And then it would also be worth me adding the MuseScore file to the repo in case future changes are needed.

What do you think?  (I know it can be very annoying when someone comes along and suggests a big change to an established workflow, so no offence if you don't like this idea!)

**Comment by aspiers** [comment]
LGTM @bennorth. Yes you got all the time signatures right, and I agree MuseScore is a fine choice for notation software.  (GNU LilyPond might deliver even higher quality output and is more amenable to the source being stored in git and to text-based peer reviews, but the learning curve is a bit steeper.) My only nitpick is that you're missing the "and" before "a partridge in a pear tree" which is needed for all days but the first.

**Comment by martinohanlon** [comment]
Labelling as low priority. This issue doesnt affect the learning outcomes of the project.

**Comment by bennorth** [comment]
Agreed on the priority!  All the same, I've created https://github.com/raspberrypilearning/code-a-carol/pull/7 with one possible set of new fragments and related changes.
